core/drivers/pl011.c

```diff
@@ -132,6 +139,13 @@ static void pl011_putc(struct serial_chip *chip, int ch)
 {
 	vaddr_t base = chip_to_base(chip);
 
+	/*
+	 * It can happen that Linux initializes the OP-TEE driver with the
+	 * console UART disabled. Avoid an infinite loop in such a case.
+	 */
+	if (!(read32(base + UART_CR) & UART_CR_UARTEN))
+		return;
+
 	/* Wait until there is space in the FIFO */
 	while (read32(base + UART_FR) & UART_FR_TXFF)
 		;
```

Comment:
What if the uart is disabled while in this loop?

Reply:
This may deserve a comment.
`/* Wait until there is space in the FIFO or device is disabled */`